### ISSUE TITLE
Is that possible to integrate with actionhero?

### ISSUE THREAD

**Issue by katopz** [OlehKulykov/FayeCpp#7]
I'm try to use FayeCpp with [actionhero](http://www.actionherojs.com/docs/core/chat.html#client-use) on my local Mac which I think I've to tear this [actionheroClient.js](https://github.com/evantahler/actionhero/blob/master/client%2FactionheroClient.js) down and convert to cpp (am I right?), Far as I can go at present is ```onFayeTransportConnected```

The question is it's possible to accomplish this? (Just to make sure I'm not wasting my time doing impossible things) Any hint is very welcome :D

Thanks

**Comment by OlehKulykov** [comment]
Hello,
I'm not so clearly understand what do you want to do. 
As I understand, you are using `actionhero` as the `faye-server`, is it right? But why do you need `actionhero javascript client`, for test? Do you need c++ client or javascript? And what part of do you want convert to c++?

**Comment by katopz** [comment]
Hi, Sorry for confusion

Yes, I'm using actionhero for faye-server (actually also primus)
and I'm has actionheroClient.js working really well via browser (which is good)

But what I need for next step is iOS and Android client, 
and I think need c++ for cross platform to save some development time.

Far as I know actionheroClient didn't have c++ provide anywhere.
So I think I should port actionheroClient js version to c++ by hand 

which FayeCpp should done 30% of work already. 
other features by primus should be another 30%.
And other chat functional should be 30%.
and something else 10% e.g. emitter and helper.

And yes I could be all wrong! Pretty need some advice here. 

FYI : Final product is cross platform chat application like HipChat target Win/Mac/iOS/Android

Thanks!

**Comment by OlehKulykov** [comment]
Hello,
now I see whole situation. 

Little bit theory: you see, any type of `faye-clients` (no mater what language they are used) is only method for communication(sending/receiving data) between client and server and not final product like chats, remote control applications, etc. 
In case of `actionhero`, they added chat functionality to the server and for the client `actionheroClient.js`, but this is not final app. 

As you wrote: `FayeCpp should done 30% of work already` - yes))), library could connect, send and receive data using Bayeux protocol as any other faye lib. But what to do with this information should decide developer(you have hummer, wood, nails, etc, but what to do next...simple chair or huge ship... is other story).

Next: as you wrote, that the `actionhero` examples is works for you in browser - than you need to look inside client `actionheroClient.js`, checkout how this chat functionality implemented and do the same. This was your own thoughts: `So I think I should port actionheroClient js version to c++ by hand`. Not port, you should implement chat functionality with selected language and platform(s).

You wrote that you targeting to `Win/Mac/iOS/Android` - than c++ is you language for this. 
I'm recommending to implement on basic chat functionality on c++ and next control it from Objective-C code or NDK or Qt(you can use it to implement single app and build it for Win/Mac, single code for the all desktop platforms).

In conclusion I want to say that you are thinking and moving in right direction and have tools for that, next combine this all together.

Happy coding....

**Comment by katopz** [comment]
Super cool! Thanks!
